### ISSUE TITLE
-ss is looking simultaneously at the same spawn point with 2 workers

### ISSUE THREAD

**Issue by Imoq** [RocketMap/RocketMap#1136]
I cover 3 small areas (-st 5) with -ss and I only turn the scan on (start the workers) when I will be around those areas. I use 2 workers for -st 5 area since I like to grab gym info as well (-gi). Since last night, I started noticing that both workers go to the exact same spawn point (exact coordinates) so, they seem to be duplicating each other's work.

I ran several tests before opening the issue, sometimes when I start the process they both attempt to go to the same point right away, and other times one worker attempts to go to one point, the second to another but after few iterations they both end up going to exactly the same one at the end.
## Expected Behavior

Each worker to go to a different spawn point!
## Current Behavior

Both workers go to the same spawn point, finding the same Pokémon and being inefficient by duplicating the work.
## Possible Solution

Make next worker to go to a different spawn point if there is one worker already on one spawn point.
## Steps to Reproduce (for bugs)

1. Start process with at least 2 accounts. 
2. Look at the log for a bit and realize both accounts go to the same spawn point.
## Context

My parameters are: "-l "43.288218, 5.380372" -st 5 -sd 10 -dc -ns -ss -gi -sn prefecture -ac accounts/prefecture.csv -w 2 -asi 10800". **PLEASE NOTICE** this behavior started before using -ac -w -asi. It most likely started yesterday.

I have some log here to provide more insight into the issue. On this example, both workers started on different spawn points, but they converged to the same time quickly. I have logs with other examples when the workers attempt to start on the exact same one since the beginning.

```
2016-09-04 19:22:50,958 [      MainThread][     runserver][    INFO] Parsed location is: 43.2804/5.3839/28.6173 (lat/lng/alt)
2016-09-04 19:22:50,985 [      MainThread][        models][    INFO] Connecting to MySQL database on localhost:3306
2016-09-04 19:22:51,079 [ search-overseer][        search][    INFO] Search overseer starting
2016-09-04 19:22:51,084 [ search-overseer][        search][    INFO] Starting account recycler thread
2016-09-04 19:22:51,093 [ search-overseer][        search][    INFO] Starting status database thread
2016-09-04 19:22:51,096 [      db-cleaner][        models][    INFO] Regular database cleaning complete
2016-09-04 19:22:51,099 [status_worker_db][        search][    INFO] Clearing previous statuses for 'perrier' worker
2016-09-04 19:22:51,102 [ search-overseer][        search][    INFO] Starting search worker threads
2016-09-04 19:22:51,117 [ search-worker-0][        search][    INFO] Waiting to get new account from the queue
2016-09-04 19:22:51,119 [ search-worker-0][        search][    INFO] Switching to account Account13
2016-09-04 19:22:51,127 [ search-overseer][        search][    INFO] New location caught, moving search grid
2016-09-04 19:22:51,129 [ search-worker-1][        search][    INFO] Waiting to get new account from the queue
2016-09-04 19:22:51,130 [ search-overseer][        models][    INFO] Finding spawn points 5 steps away
2016-09-04 19:22:51,130 [ search-worker-1][        search][    INFO] Switching to account Account14
2016-09-04 19:22:52,556 [ search-overseer][        search][    INFO] Total of 337 spawns to track
2016-09-04 19:22:52,578 [ search-worker-0][        search][    INFO] Early for 43.282153,5.389824; waiting 11s...
2016-09-04 19:22:52,579 [ search-worker-1][        search][    INFO] Early for 43.280785,5.386880; waiting 18s...
2016-09-04 19:23:03,592 [ search-worker-0][        search][    INFO] Searching at 43.282153,5.389824
2016-09-04 19:23:03,594 [ search-worker-0][      auth_ptc][    INFO] PTC User Login for: Account13
2016-09-04 19:23:05,615 [ search-worker-0][      auth_ptc][    INFO] PTC User Login successful.
2016-09-04 19:23:05,616 [ search-worker-0][      auth_ptc][    INFO] Request PTC Access Token...
2016-09-04 19:23:05,815 [ search-worker-0][      auth_ptc][    INFO] PTC Access Token successfully retrieved.
2016-09-04 19:23:10,608 [ search-worker-1][        search][    INFO] Searching at 43.280785,5.386880
2016-09-04 19:23:10,611 [ search-worker-1][      auth_ptc][    INFO] PTC User Login for: Account14
2016-09-04 19:23:13,028 [ search-worker-1][      auth_ptc][    INFO] PTC User Login successful.
2016-09-04 19:23:13,028 [ search-worker-1][      auth_ptc][    INFO] Request PTC Access Token...
2016-09-04 19:23:13,424 [ search-worker-1][      auth_ptc][    INFO] PTC Access Token successfully retrieved.
======================================
 Name: spearow
 Rarity: uncommon
 Coord: (43.281863,5.389579)
 ID: 21
 Remaining Time: 0:11:23.737222
======================================
2016-09-04 19:23:18,590 [ search-worker-0][        models][    INFO] Parsing found 1 pokemons, 164 pokestops, and 16 gyms
2016-09-04 19:23:22,100 [ search-worker-0][        models][    INFO] Upserted 1 gyms and 8 gym members
======================================
 Name: staryu
 Rarity: uncommon
 Coord: (43.280785,5.386880)
 ID: 120
 Remaining Time: 0:14:35.084884
======================================
======================================
 Name: weedle
 Rarity: common
 Coord: (43.280706,5.386961)
 ID: 13
 Remaining Time: 0:03:19.660445
======================================
======================================
 Name: pidgey
 Rarity: common
 Coord: (43.280901,5.387616)
 ID: 16
 Remaining Time: 0:08:32.660186
======================================
2016-09-04 19:23:25,570 [ search-worker-1][        models][    INFO] Parsing found 3 pokemons, 152 pokestops, and 16 gyms
2016-09-04 19:23:32,112 [ search-worker-0][        search][    INFO] Searching at 43.280785,5.386880
======================================
 Name: staryu
 Rarity: uncommon
 Coord: (43.280785,5.386880)
 ID: 120
 Remaining Time: 0:14:27.810045
======================================
======================================
 Name: weedle
 Rarity: common
 Coord: (43.280706,5.386961)
 ID: 13
 Remaining Time: 0:03:12.385087
======================================
======================================
 Name: pidgey
 Rarity: common
 Coord: (43.280901,5.387616)
 ID: 16
 Remaining Time: 0:08:25.384721
======================================
2016-09-04 19:23:32,843 [ search-worker-0][        models][    INFO] Parsing found 3 pokemons, 156 pokestops, and 17 gyms
2016-09-04 19:23:35,917 [ search-worker-1][        search][    INFO] Searching at 43.284671,5.382708
======================================
 Name: venonat
 Rarity: uncommon
 Coord: (43.285029,5.383117)
 ID: 48
 Remaining Time: 0:10:24.012582
======================================
======================================
 Name: pidgey
 Rarity: common
 Coord: (43.284973,5.382708)
 ID: 16
 Remaining Time: 0:02:33.932133
======================================
======================================
 Name: caterpie
 Rarity: uncommon
 Coord: (43.284969,5.382790)
 ID: 10
 Remaining Time: 0:14:56.083850
======================================
======================================
 Name: mr. mime
 Rarity: very rare
 Coord: (43.284671,5.382708)
 ID: 122
 Remaining Time: 0:14:47.819618
======================================
2016-09-04 19:23:36,729 [ search-worker-1][        models][    INFO] Parsing found 4 pokemons, 228 pokestops, and 16 gyms
2016-09-04 19:23:39,823 [ search-worker-1][        models][    INFO] Upserted 1 gyms and 2 gym members
2016-09-04 19:23:43,150 [ search-worker-0][        search][    INFO] Searching at 43.284671,5.382708
======================================
 Name: venonat
 Rarity: uncommon
 Coord: (43.285029,5.383117)
 ID: 48
 Remaining Time: 0:10:16.746785
======================================
======================================
 Name: pidgey
 Rarity: common
 Coord: (43.284973,5.382708)
 ID: 16
 Remaining Time: 0:02:26.665778
======================================
======================================
 Name: caterpie
 Rarity: uncommon
 Coord: (43.284969,5.382790)
 ID: 10
 Remaining Time: 0:14:48.817441
======================================
======================================
 Name: mr. mime
 Rarity: very rare
 Coord: (43.284671,5.382708)
 ID: 122
 Remaining Time: 0:14:40.553186
======================================
2016-09-04 19:23:43,996 [ search-worker-0][        models][    INFO] Parsing found 4 pokemons, 228 pokestops, and 16 gyms
2016-09-04 19:23:47,255 [ search-worker-0][        models][    INFO] Upserted 1 gyms and 1 gym members
2016-09-04 19:23:49,834 [ search-worker-1][        search][    INFO] Searching at 43.284969,5.382790
======================================
 Name: abra
 Rarity: rare
 Coord: (43.285401,5.383199)
 ID: 63
 Remaining Time: 0:09:14.999160
======================================
======================================
 Name: venonat
 Rarity: uncommon
 Coord: (43.285029,5.383117)
 ID: 48
 Remaining Time: 0:10:10.038002
======================================
======================================
 Name: pidgey
 Rarity: common
 Coord: (43.284973,5.382708)
 ID: 16
 Remaining Time: 0:02:19.957637
======================================
======================================
 Name: caterpie
 Rarity: uncommon
 Coord: (43.284969,5.382790)
 ID: 10
 Remaining Time: 0:14:42.109407
======================================
======================================
 Name: mr. mime
 Rarity: very rare
 Coord: (43.284671,5.382708)
 ID: 122
 Remaining Time: 0:14:33.845186
======================================
2016-09-04 19:23:50,704 [ search-worker-1][        models][    INFO] Parsing found 5 pokemons, 236 pokestops, and 16 gyms
2016-09-04 19:23:51,155 [account-recycler][        search][    INFO] Account recycler running. Checking status of 0 accounts
2016-09-04 19:23:51,170 [      db-cleaner][        models][    INFO] Regular database cleaning complete
2016-09-04 19:23:57,267 [ search-worker-0][        search][    INFO] Searching at 43.284969,5.382790
======================================
 Name: abra
 Rarity: rare
 Coord: (43.285401,5.383199)
 ID: 63
 Remaining Time: 0:09:07.402936
======================================
======================================
 Name: venonat
 Rarity: uncommon
 Coord: (43.285029,5.383117)
 ID: 48
 Remaining Time: 0:10:02.441770
======================================
======================================
 Name: pidgey
 Rarity: common
 Coord: (43.284973,5.382708)
 ID: 16
 Remaining Time: 0:02:12.361425
======================================
======================================
 Name: caterpie
 Rarity: uncommon
 Coord: (43.284969,5.382790)
 ID: 10
 Remaining Time: 0:14:34.513179
======================================
======================================
 Name: mr. mime
 Rarity: very rare
 Coord: (43.284671,5.382708)
 ID: 122
 Remaining Time: 0:14:26.248883
======================================
2016-09-04 19:23:58,300 [ search-worker-0][        models][    INFO] Parsing found 5 pokemons, 236 pokestops, and 16 gyms
2016-09-04 19:24:01,154 [ search-worker-1][        search][    INFO] Early for 43.284056,5.384590; waiting 23s...
2016-09-04 19:24:08,725 [ search-worker-0][        search][    INFO] Early for 43.284056,5.384590; waiting 16s...
2016-09-04 19:24:24,187 [ search-worker-1][        search][    INFO] Searching at 43.284056,5.384590
2016-09-04 19:24:24,740 [ search-worker-0][        search][    INFO] Searching at 43.284056,5.384590
======================================
 Name: bulbasaur
 Rarity: uncommon
 Coord: (43.284056,5.384590)
 ID: 1
 Remaining Time: 0:14:49.846091
======================================
```

After that point, all next search attempts are being made to the same coordinates.
## Your Environment

- Version used:
- Environment name and version (e.g. Python 2.7):
- Operating System and version (desktop or mobile):

Raspberry Pi, Python 2.7, latest devel commit:

```
root@raspberrypi:~/PokemonGo-Map# git log | head
commit d92e8eedadf46cf25eeea0e4ca88313ed99918c2
Author: FrostTheFox <frost@walnutgaming.com>
Date:   Sat Sep 3 23:32:17 2016 -0400

    Dumping feathub, add new FR info

commit ed1ee704321450aa7d05af42e1e2f3e55a0cf708
Author: justsomedudeonthenet <tizul@divismail.ru>
Date:   Sat Sep 3 20:39:31 2016 -0400
```

**Comment by Imoq** [comment]
Another example of starting a process and both workers go straight to the same point since the beginning:

```
2016-09-04 19:55:08,165 [      MainThread][     runserver][    INFO] Parsed location is: 43.2804/5.3839/28.6173 (lat/lng/alt)
2016-09-04 19:55:08,189 [      MainThread][        models][    INFO] Connecting to MySQL database on localhost:3306
2016-09-04 19:55:08,283 [ search-overseer][        search][    INFO] Search overseer starting
2016-09-04 19:55:08,289 [ search-overseer][        search][    INFO] Starting account recycler thread
2016-09-04 19:55:08,298 [ search-overseer][        search][    INFO] Starting status database thread
2016-09-04 19:55:08,306 [      db-cleaner][        models][    INFO] Regular database cleaning complete
2016-09-04 19:55:08,308 [status_worker_db][        search][    INFO] Clearing previous statuses for 'perrier' worker
2016-09-04 19:55:08,313 [ search-overseer][        search][    INFO] Starting search worker threads
2016-09-04 19:55:08,315 [ search-worker-0][        search][    INFO] Waiting to get new account from the queue
2016-09-04 19:55:08,318 [ search-worker-0][        search][    INFO] Switching to account Account13
2016-09-04 19:55:08,323 [ search-overseer][        search][    INFO] New location caught, moving search grid
2016-09-04 19:55:08,325 [ search-overseer][        models][    INFO] Finding spawn points 5 steps away
2016-09-04 19:55:08,326 [ search-worker-1][        search][    INFO] Waiting to get new account from the queue
2016-09-04 19:55:08,339 [ search-worker-1][        search][    INFO] Switching to account Account14
2016-09-04 19:55:09,774 [ search-overseer][        search][    INFO] Total of 339 spawns to track
2016-09-04 19:55:09,796 [ search-worker-0][        search][    INFO] Early for 43.283546,5.382627; waiting 16s...
2016-09-04 19:55:09,798 [ search-worker-1][        search][    INFO] Early for 43.283546,5.382627; waiting 16s...
^CTraceback (most recent call last):

```

**Comment by Albert62** [comment]
Really annoying..

**Comment by Cartman123** [comment]
Are you using sqlite or mysql?

**Comment by Imoq** [comment]
@Cartman123 MySQL:

```
2016-09-04 19:22:50,958 [      MainThread][     runserver][    INFO] Parsed location is: 43.2804/5.3839/28.6173 (lat/lng/alt)
2016-09-04 19:22:50,985 [      MainThread][        models][    INFO] Connecting to MySQL database on localhost:3306
```

I was using SQLite few days ago and I converted the database to MySQL. Does this have something to do with it? Makes sense, I think it started this way since I converted, now that I remember...

**Comment by Cartman123** [comment]
Nevermind, it says on you log you're using MySQL.

on pogom/models.py look for this code:

```
 # Sqlite doesn't support distinct on columns
        if args.db_type == 'mysql':
            query = query.distinct(Pokemon.spawnpoint_id)
        else:
            query = query.group_by(Pokemon.spawnpoint_id)
```

and replace it with this:

```
 # Sqlite doesn't support distinct on columns
        if args.db_type == 'mysql':
#            query = query.distinct(Pokemon.spawnpoint_id)
            query = query.group_by(Pokemon.spawnpoint_id)
        else:
            query = query.group_by(Pokemon.spawnpoint_id)
```

Let me know if it fixes it. It fixed mine. It seems distinct isn't really working for some reason.

**Comment by Imoq** [comment]
@Cartman123 looks like it did the trick! I have only been running it for 3 minutes now, but all scanned points have been different so far. I'll report back if I observe anything wrong, but I bet your fix did it :) Thank you!

**Comment by Cartman123** [comment]
@Imoq Glad it worked out :)

I'm not a mantainer of this project and I'm not that familiar with the code, so I can't say the performance impact it has overall, or if it was deliberate, so I won't send a PR for this. Someone pointed out on another issue that distinct isn't really THAT distinct, but it's worth it for perfomance.

Maybe some mantainer can shed some light on this?

**Comment by Imoq** [comment]
Understood.

I hope this gets the attention of a maintainer, seems like a big issue for sure. I don't know if it only happens if you migrated your database from SQLite to MySQL or if it happens all the time.

**Comment by justsomedudeonthenet** [comment]
If you are still able to cause this behaviour, could you please run with '-vv logfile.txt'?

Particularly I want to see the section where it dumps out all the spawnpoints it is going to scan.  Should have a bunch of lines with that time, lat and lng, right after "Total of x spawns to track"

**Comment by Imoq** [comment]
@justsomedudeonthenet : 

I was able to easily reproduce it by going back to the original models.py. I believe the section you asked for is this (only pasting the very start of the section, but I believe you can see the repeated spawn points there. Strange that it seems like not all of them are repeated, no?):

```
2016-09-07 02:31:20,686 [ search-overseer][          pool][   DEBUG] No connection available in pool.
2016-09-07 02:31:20,692 [    db-updater-0][        models][   DEBUG] Upserted to WorkerStatus, 2 records (upsert queue remaining: 0)
2016-09-07 02:31:20,696 [ search-overseer][          pool][   DEBUG] Created new connection 1968130960.
2016-09-07 02:31:20,924 [      MainThread][      __init__][   DEBUG] Finished computing cache-busting values
2016-09-07 02:31:21,496 [ search-worker-1][        pgoapi][    INFO] pgoapi v1.1.7 - Copyright (c) 2016 tjado <https://github.com/tejado>
2016-09-07 02:31:22,298 [ search-overseer][        search][    INFO] Total of 376 spawns to track
2016-09-07 02:31:22,301 [ search-overseer][        search][   DEBUG] Scan [00:00] (0) @ 43.2921406172,5.37919173091
2016-09-07 02:31:22,301 [ search-overseer][        search][   DEBUG] Scan [00:00] (0) @ 43.2921406172,5.37919173091
2016-09-07 02:31:22,302 [ search-overseer][        search][   DEBUG] Scan [00:12] (12) @ 43.2923296313,5.37837392403
2016-09-07 02:31:22,303 [ search-overseer][        search][   DEBUG] Scan [00:30] (30) @ 43.286140005,5.37870104539
2016-09-07 02:31:22,304 [ search-overseer][        search][   DEBUG] Scan [00:30] (30) @ 43.286140005,5.37870104539
2016-09-07 02:31:22,305 [ search-overseer][        search][   DEBUG] Scan [00:40] (40) @ 43.291580557,5.37829214397
2016-09-07 02:31:22,305 [ search-overseer][        search][   DEBUG] Scan [00:40] (40) @ 43.291580557,5.37829214397
2016-09-07 02:31:22,306 [ search-overseer][        search][   DEBUG] Scan [00:49] (49) @ 43.2905738663,5.38368987565
2016-09-07 02:31:22,307 [ search-overseer][        search][   DEBUG] Scan [01:08] (68) @ 43.2845194863,5.37632945747
2016-09-07 02:31:22,308 [ search-overseer][        search][   DEBUG] Scan [01:08] (68) @ 43.2845194863,5.37632945747
2016-09-07 02:31:22,308 [ search-overseer][        search][   DEBUG] Scan [01:24] (84) @ 43.2923296313,5.37837392403
2016-09-07 02:31:22,309 [ search-overseer][        search][   DEBUG] Scan [01:54] (114) @ 43.2881154221,5.37992776699
2016-09-07 02:31:22,310 [ search-overseer][        search][   DEBUG] Scan [01:54] (114) @ 43.2881154221,5.37992776699
2016-09-07 02:31:22,311 [ search-overseer][        search][   DEBUG] Scan [01:54] (114) @ 43.2926083833,5.38205414614
2016-09-07 02:31:22,311 [ search-overseer][        search][   DEBUG] Scan [01:54] (114) @ 43.2926083833,5.38205414614
2016-09-07 02:31:22,312 [ search-overseer][        search][   DEBUG] Scan [02:27] (147) @ 43.2923296313,5.37837392403
2016-09-07 02:31:22,313 [ search-overseer][        search][   DEBUG] Scan [03:13] (193) @ 43.2905738663,5.38368987565
2016-09-07 02:31:22,314 [ search-overseer][        search][   DEBUG] Scan [03:25] (205) @ 43.2905738663,5.38368987565
2016-09-07 02:31:22,314 [ search-overseer][        search][   DEBUG] Scan [03:25] (205) @ 43.2905738663,5.38368987565
2016-09-07 02:31:22,315 [ search-overseer][        search][   DEBUG] Scan [04:14] (254) @ 43.2922159085,5.37919173091
2016-09-07 02:31:22,316 [ search-overseer][        search][   DEBUG] Scan [04:14] (254) @ 43.2922159085,5.37919173091
2016-09-07 02:31:22,317 [ search-overseer][        search][   DEBUG] Scan [04:21] (261) @ 43.2848044834,5.38148165175
2016-09-07 02:31:22,317 [ search-overseer][        search][   DEBUG] Scan [04:21] (261) @ 43.2848044834,5.38148165175
2016-09-07 02:31:22,318 [ search-overseer][        search][   DEBUG] Scan [04:41] (281) @ 43.2917157697,5.37861926487
2016-09-07 02:31:22,319 [ search-overseer][        search][   DEBUG] Scan [04:41] (281) @ 43.2917157697,5.37861926487
2016-09-07 02:31:22,320 [ search-overseer][        search][   DEBUG] Scan [04:48] (288) @ 43.2918769182,5.38319915193
2016-09-07 02:31:22,320 [ search-overseer][        search][   DEBUG] Scan [04:48] (288) @ 43.2918769182,5.38319915193
2016-09-07 02:31:22,321 [ search-overseer][        search][   DEBUG] Scan [04:59] (299) @ 43.2918025905,5.37837392403
2016-09-07 02:31:22,322 [ search-overseer][        search][   DEBUG] Scan [04:59] (299) @ 43.2918025905,5.37837392403
2016-09-07 02:31:22,323 [ search-overseer][        search][   DEBUG] Scan [05:21] (321) @ 43.2919816156,5.37616590327
2016-09-07 02:31:22,323 [ search-overseer][        search][   DEBUG] Scan [05:21] (321) @ 43.2919816156,5.37616590327
2016-09-07 02:31:22,324 [ search-overseer][        search][   DEBUG] Scan [05:25] (325) @ 43.2923296313,5.37837392403
2016-09-07 02:31:22,325 [ search-overseer][        search][   DEBUG] Scan [05:26] (326) @ 43.2923296313,5.37837392403
2016-09-07 02:31:22,325 [ search-overseer][        search][   DEBUG] Scan [05:39] (339) @ 43.2921703898,5.38336272604
2016-09-07 02:31:22,326 [ search-overseer][        search][   DEBUG] Scan [05:39] (339) @ 43.2921703898,5.38336272604
2016-09-07 02:31:22,327 [ search-overseer][        search][   DEBUG] Scan [05:47] (347) @ 43.2882957785,5.38409881526
2016-09-07 02:31:22,328 [ search-overseer][        search][   DEBUG] Scan [05:47] (347) @ 43.2882957785,5.38409881526
2016-09-07 02:31:22,329 [ search-overseer][        search][   DEBUG] Scan [05:50] (350) @ 43.2905617297,5.38074559587
2016-09-07 02:31:22,329 [ search-overseer][        search][   DEBUG] Scan [05:50] (350) @ 43.2905617297,5.38074559587
```

I have the complete debug file (480 kb for 1-2 minutes of run!) if you need me to upload it somewhere.

Thank you for looking at this!

**Comment by Imoq** [comment]
"I R stooopid", I didn't see the #1179  fix before reporting back. Applying in a second and letting you know :S Sorry!

**Comment by justsomedudeonthenet** [comment]
@Imoq that is exactly what I wanted to see, thanks.  And now we know for sure that PR fixes it.  Excellent.

I will see what we can do about getting it merged quickly.

**Comment by averylarry** [comment]
I think maybe there are 3 separate things happening.
1)  Inconsistent data convention.  Looks like over time not all lat/lon used the same number of decimal places.
2)  Inconsistent data.  In my database, spawnpoints can have multiple different disappear times, often only different by less than 10 seconds.  I don't know how it happened.  This doesn't seem right to me but maybe it is?  Needs investigation.  Note that I cleared out my database to see if I can maintain consistent data going forward.
3)  Poor query construct.  I don't believe that group by spawnpoint_id is appropriate.  We want unique sets of latitude, longitude, and hourly disappear time.  I don't know enough about MySQL, SQLite, nor python to change it.  In SQL I would:
select latitude, longitude, (disappear time minute*60 + disappear time seconds) time
from pokemon
group by lattitude, longitude, (disappear time minute*60 + disappear time seconds)

Alternatively, if you want to attempt to bypass 1 instead of fixing it, you could in SQL:
select max(latitude) latitude, max(longitude) longitude, (disappear time minute*60 + disappear time seconds) time
from pokemon
group by spawnpoint_id, (disappear time minute*60 + disappear time seconds)

Neither of those addresses 2.

**Comment by averylarry** [comment]
@justsomedudeonthenet 
The 595 fix from #572 is populating (in my opinion) incorrect disappear times for the non-standard 15 minute spawnpoints (if I understand the code, it just puts in 15 minutes from the time it was first seen).

https://github.com/PokemonGoMap/PokemonGo-Map/blob/e92fa833745e301af88eb8e1f5e592c7d3283291/pogom/models.py#L593

This is creating (again in my opinion) incorrect spawnpoint information in the spawnpoint dumps for -ss scanning.  I suppose it's fairly minor.  There will be extra spawnpoint/spawn time combinations dumped and scanned.  Compression algorithms are likely to cluster these together fairly well.

Perhaps instead an arbitrary datetime could be used to indicate unknown disappear times (like 1900-01-01 or 3000-01-0).  This datetime could be excluded from relevant queries (like spawnpoint dumps).
